packages/shared/test/unit/args_tests.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-env jest */
+import { args, getArgName } from '../../args';
+
+describe('args(args)', () => {
+    it('it should parse the arguments correctly', () => {
+        const input = ['hello', 'world', '-c', 'config', '--filename', 'filename', 'otherargs', 'remaining'];
+        const expectedOutput = {
+            named: { c: 'config', filename: 'filename' },
+            positional: ['hello', 'world', 'otherargs', 'remaining']
+        };
+        expect(args(input)).toEqual(expectedOutput);
+    });
+
+    it('it should throw an error if value is not passed for arg', () => {
+        const input = ['hello', 'world', '-c'];
+        expect(() => args(input)).toThrowError();
+    });
+});
+
+describe('getArgName(arg)', () => {
+    it('it should extract the argument name for single character arg', () => {
```

Comment:
This test states that a '-c' is a single character argument. This is a good condition to have, but it is not enforced in the code. We should make the code throw an Error when the argument is prefixed with a single '-' but the name is longer that one character.

Reply:
I've added those checks.